File: bank-service/src/main/java/rs/edu/raf/model/card/CardIssuer.java
package rs.edu.raf.model.card;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Entity
@Getter
@Setter
public class CardIssuer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(unique = true)
    private String name;
    private int MII;
    private int BIN;
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY, mappedBy = "cardIssuer")
    private List<Card> cards;
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/model/card/CardIssuer.kt
package org.example.bankservicekotlin.model.card

import jakarta.persistence.*

@Entity
class CardIssuer(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long?,
    @Column(unique = true)
    val name: String,
    val MII: Int,
    val BIN: Int,
    @OneToMany(cascade = [CascadeType.ALL], orphanRemoval = true, fetch = FetchType.LAZY, mappedBy = "cardIssuer")
    val cards: MutableList<Card>?
) {
}